packages/xrpl/src/Wallet/index.ts

```diff
@@ -197,6 +198,38 @@ class Wallet {
     })
   }
 
+  /**
+   * Derives a wallet from secret numbers.
+   *
+   * @param secretNumbers - A string consisting of 8 times 6 numbers (whitespace delimited) used to derive a wallet.
+   * @param opts - (Optional) Options to derive a Wallet.
+   * @param opts.masterAddress - Include if a Wallet uses a Regular Key Pair. It must be the master address of the account.
+   * @param opts.algorithm - The digital signature algorithm to generate an address for.
+   * @returns A Wallet derived from secret numbers.
+   * @throws ValidationError if unable to derive private key from secret number input.
+   */
+  public static fromSecretNumbers(
+    secretNumbers: string[] | string,
+    opts: { masterAddress?: string; algorithm?: ECDSA } = {},
+  ): Wallet {
+    let numbersArray: string[] = []
+
+    if (typeof secretNumbers === 'string') {
+      numbersArray = Utils.parseSecretString(secretNumbers)
+    } else if (Array.isArray(secretNumbers)) {
+      numbersArray = secretNumbers
+    } else {
+      throw new ValidationError('Unable to parse the given secret numbers.')
+    }
+
+    const entropy = Utils.secretToEntropy(numbersArray)
+
+    return Wallet.fromEntropy(entropy, {
+      algorithm: opts.algorithm ?? ECDSA.secp256k1,
```

Comment:
should use `DEFAULT_ALGORITHM` as a fallback here, not `secp256k1`

Reply:
you can also just forward the `opts` dict entirely:

```javascript
return Wallet.fromEntropy(entropy, opts)
```

Reply:
I decided to fallback to `secp256k1` because, the only wallet using "Secret Numbers" so far is XUMM. And they use `secp256k1` and not `ed25519`.

So if someone wants to create a xrpl.js "Wallet.fromSecretNumbers" he always have to define the algorithm `secp256k1` or else he ends up with a different address.

Reply:
That makes sense, could you add a comment with that context?

Reply:
Yes, will do tomorrow 🙂

Reply:
Sorry I don't agree. I think that XUMM in this case should override the default. We want this lib to be useful to anyone and we want people to default to ed25519

Reply:
I meant it the other way around: Users create a wallet in XUMM. This process gives them the "Secret Numbers" via `secp256k1`.

Now this XUMM user is also a developer. So he wants to build on XRPL, using xrpl.js and his Secret Numbers he got from XUMM. If `ed25519` is the default, the user always has to specify the algorithm when using the ".fromSecretNumbers" method.

What is the reason to always default to ed25519?

Reply:
The reason to default to ed25519 is that it's apparently got better performance. (See comment on the two algorithms here: https://xrpl.org/cryptographic-keys.html#signing-algorithms)